datamodel/include/data_types/rendering/RenderingContext.hpp

```diff
@@ -28,32 +32,72 @@ struct RenderingContext {
     RenderingContext &operator=(RenderingContext&& other) = delete;
 
     void gl_context_from_this() const {
-        auto status = glXMakeCurrent(x_display, gl_x_window, gl_x_context);
+        auto status = glXMakeCurrent(x_display_, gl_x_window_, gl_x_context_);
         if (status == False) {
             throw std::runtime_error("Can't set rendering context from current GLX context");
         }
     }
 
     void gl_context_from_default() const {
-        auto status = glXMakeCurrent(x_display, None, nullptr);
+        auto status = glXMakeCurrent(x_display_, None, nullptr);
         if (status == False) {
             throw std::runtime_error("Can't set rendering context to default value");
         }
     }
 
+    void set_display(Display* d);
+    [[nodiscard]] Display* get_display() const;
+
+    void set_delete_message(Atom a);
+    [[nodiscard]] Atom get_delete_message() const;
+
+    void set_glx_context(GLXContext glxc);
+    [[nodiscard]] GLXContext get_glx_context() const;
+
+    void set_drawable(GLXDrawable d);
+    [[nodiscard]] GLXDrawable get_drawable() const;
+
+    void set_background_color(const rcbe::visual::RGBAColor& color);
+    [[nodiscard]] const rcbe::visual::RGBAColor& get_background_color() const;
+
+    void set_window_dimensions(const rcbe::core::Dimensions& d);
+    [[nodiscard]] const rcbe::core::Dimensions& get_window_dimensions() const;
+
+    void update_transform(const rcbe::math::Transform& trn);
+    [[nodiscard]] const math::MatrixColumnMajorAdaptor& get_transform_column_major() const;
+
+    void set_mouse_coordinates(const rcbe::math::Vector2d& v);
+    [[nodiscard]] const std::optional<rcbe::math::Vector2d>& get_mouse_coordinates() const;
+
+    [[nodiscard]] rcbe::core::EngineScalar get_zoom() const;
+
+    void zoom_in(rcbe::core::EngineScalar step = 1.);
```

Comment:
This kind of puzzles me, I don't think that zooming in or out is a `RenderingContext`'s job, add a simple setter here, and relocate these to `Camera`.